packages/sw-lib/src/lib/router.js

```diff
@@ -112,14 +121,23 @@ class Router extends SWRoutingRouter {
    * @param {Array<RegExp>} [options.blacklist] Defaults to an empty blacklist.
    * @param {Array<RegExp>} [options.whitelist] Defaults to `[/./]`, which will
    *        match all request URLs.
+   * @param {String} [options.cacheName] The name of the cache which contains
+   *        the cached response for `url`. Defaults to the name of the cache
+   *        used by cacheRevisionedAssets().
    */
   registerNavigationRoute(url, options = {}) {
     if (typeof url !== 'string') {
       throw ErrorFactory.createError('navigation-route-url-string');
     }
 
+    // Allow folks to explicitly pass in a null/undefined cacheName option if
+    // they want that behavior.
+    const cacheName = 'cacheName' in options ?
```

Comment:
What's the use-case for specifying a null or undefined cacheName? (I may be missing something obvious). Is it for the case where folks want to treat options.cacheName as optional and ignore setting a name themselves?